content/en/ebpf.md

```diff
@@ -0,0 +1,36 @@
+---
+title: eBPF
+status: Completed
+category: architecture
+---
+
+## What it is
+
+eBPF, or extended Berkeley Packet Filter, is a technology that allows small, sandboxed programs or scripts to be run in the kernel space of a Linux system without having to change the kernel's source code or load Linux kernel modules.
+
+A Linux system has two spaces: the kernel and the user space. 
+The kernel represents the operating system's core and is the only part with unlimited access to the hardware. 
+
```

Comment:
From my personal taste, we go too much into detail in the following sentences, especially as we talk about some of those points in the next chapter. What do you think? Maybe we can shrink the next sentences to just add something like: 
Applications reside in the user space but sometimes require deeper access to the operating system than the default kernel provides. Linux kernel modules can extend the kernel's default functionality allowing applications to get deeper access to underlying components. Unfortunately, this could introduce security risks as well.

Reply:
Thanks, @iamNoah1 for your suggestions.

I understand your point here and I agree that it has become long and what you suggested summarises it. However, the first time that I encountered the term, it was pretty hard for me personally to understand it. That's why I tried to break it down into smaller and simpler definitions. What do you think if we mix the two, making it short, but still explaining a bit more? Something like:

> Applications reside in the user space, and when they need higher permissions, they send a request to the kernel. 
For applications that require more flexibility, such as direct hardware access, the kernel can be extended via what is known as the "Linux kernel modules" approach. It can extend the kernel's default functionality allowing applications deeper access to the underlying components. Unfortunately, this could introduce security risks as well.

Reply:
@MaryamTavakkoli Yes that would be fine from my side :)

Reply:
I am confused, did you apply the suggestions?!